compiler-plugin/src/main/kotlin/com/google/devtools/ksp/KotlinSymbolProcessingExtension.kt

```diff
@@ -198,6 +204,42 @@ abstract class AbstractKotlinSymbolProcessingExtension(
             newFiles, deferredSymbols, bindingTrace, project, componentProvider, incrementalContext, options
         )
 
+        if (!initialized) {
+            // Visit constants so that JavaPsiFacade knows them.
+            // The annotation visitor in ResolverImpl covered newFiles already.
+            // Skip private and local members, which are not visible to Java files.
+            ksFiles.filterIsInstance<KSFileImpl>().filter { it !in dirtyFiles }.forEach {
+                try {
+                    it.accept(
+                        object : KSVisitorVoid() {
+                            private fun visitDeclarationContainer(container: KSDeclarationContainer) {
+                                container.declarations.filterNot {
+                                    it.getVisibility() == Visibility.PRIVATE
+                                }.forEach {
+                                    it.accept(this, Unit)
+                                }
+                            }
+
+                            override fun visitFile(file: KSFile, data: Unit) =
+                                visitDeclarationContainer(file)
+
+                            override fun visitClassDeclaration(classDeclaration: KSClassDeclaration, data: Unit) =
+                                visitDeclarationContainer(classDeclaration)
+
+                            override fun visitPropertyDeclaration(property: KSPropertyDeclaration, data: Unit) {
+                                if (property.modifiers.contains(Modifier.CONST)) {
+                                    property.getter // force resolution
+                                }
+                            }
+                        },
+                        Unit
+                    )
+                } catch (_: Exception) {
```

Comment:
what's the case for exception here and why it is simply ignored for the general `Exception` rather than more specific exception types?

Reply:
It's similar to the initialization blocks protected by `handleException` in the above. Exceptions can arise from, e.g., broken inputs. This is the best effort to make constants in Kotlin available to JavaPsiFacade and try not to error out if possible. It's up to the processors to decide whether to error out or not.